src/cpp/src/whisper/whisper_models.hpp

```diff
@@ -0,0 +1,19 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <openvino/openvino.hpp>
+
+namespace ov {
+namespace genai {
+
+struct WhisperInitializedModels {
+    ov::InferRequest encoder;
+    ov::CompiledModel decoder_compiled;
```

Comment:
do we need to store both CompiledModel and InferRequest? typically, infer request holds internally pointer to compiled model

Reply:
Thanks, switched to use request pointer. Further improvement is to cache key, value tensor names and not to query compiled model outputs at all.